tests/modules/images/regenerate-existing-images/background-process/perflab-background-job-test.php

```diff
@@ -0,0 +1,173 @@
+ * @group regenerate-existing-images
+ */
+class Perflab_Background_Job_Test extends WP_UnitTestCase {
+
+	public function test_class_constants_exists() {
+		$job_class = Perflab_Background_Job::class;
+
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_ACTION' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_DATA' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_ATTEMPTS' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_LOCK' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_ERRORS' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_STATUS' ) );
+		$this->assertTrue( defined( $job_class . '::META_KEY_JOB_COMPLETED_AT' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_QUEUED' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_RUNNING' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_PARTIAL' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_COMPLETE' ) );
+		$this->assertTrue( defined( $job_class . '::JOB_STATUS_FAILED' ) );
+	}
+
+	public function test_class_constants_values() {
+		$job_class = Perflab_Background_Job::class;
+
+		$this->assertEquals( 'perflab_job_action', constant( $job_class . '::META_KEY_JOB_ACTION' ) );
+		$this->assertEquals( 'perflab_job_data', constant( $job_class . '::META_KEY_JOB_DATA' ) );
+		$this->assertEquals( 'perflab_job_attempts', constant( $job_class . '::META_KEY_JOB_ATTEMPTS' ) );
+		$this->assertEquals( 'perflab_job_lock', constant( $job_class . '::META_KEY_JOB_LOCK' ) );
+		$this->assertEquals( 'perflab_job_errors', constant( $job_class . '::META_KEY_JOB_ERRORS' ) );
+		$this->assertEquals( 'perflab_job_status', constant( $job_class . '::META_KEY_JOB_STATUS' ) );
+		$this->assertEquals( 'perflab_job_completed_at', constant( $job_class . '::META_KEY_JOB_COMPLETED_AT' ) );
+		$this->assertEquals( 'queued', constant( $job_class . '::JOB_STATUS_QUEUED' ) );
+		$this->assertEquals( 'running', constant( $job_class . '::JOB_STATUS_RUNNING' ) );
+		$this->assertEquals( 'partial', constant( $job_class . '::JOB_STATUS_PARTIAL' ) );
+		$this->assertEquals( 'completed', constant( $job_class . '::JOB_STATUS_COMPLETE' ) );
+		$this->assertEquals( 'failed', constant( $job_class . '::JOB_STATUS_FAILED' ) );
+	}
+
+	/**
+	 * @covers ::set_status
+	 */
+	public function test_set_status_false_for_invalid_status() {
+		$job = perflab_create_background_job( 'test', array( 'action' => 'testing_suite' ) );
+
+		$result = $job->set_status( 'invalid_status' );
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @covers ::set_status
+	 * @dataProvider job_provider
+	 */
+	public function test_set_status_false_for_valid_status( $job ) {
+		$this->assertTrue( $job->set_status( 'running' ) );
+		$this->assertTrue( $job->set_status( 'partial' ) );
+		$this->assertTrue( $job->set_status( 'failed' ) );
+		$this->assertTrue( $job->set_status( 'completed' ) );
+	}
+
```

Comment:
In my comment https://github.com/WordPress/performance/pull/507#discussion_r975857372 I didn't mean to use a data provider for the job. A data provider should be used for the values which would be different for otherwise almost identical tests.

So the idea here would be to have a data provider which returns the array of statuses, so that way you could change this method to a simple `test_set_status_false_for_valid_status( $status )` with only a single assertion.

Reply:
@ankitrox Left two follow-up comments in https://github.com/WordPress/performance/pull/507#pullrequestreview-1126123110, this still doesn't look right.